packages/kit/src/runtime/server/page/index.js

```diff
@@ -70,7 +70,9 @@ export async function render_page(event, page, options, manifest, state, resolve
 			}
 		}
 
-		const should_prerender_data = nodes.some((node) => node?.server?.load);
+		const should_prerender_data = nodes.some(
+			(node) => node?.server?.load || node?.server?.trailingSlash !== undefined
```

Comment:
```suggestion
			// prerender in case of trailingSlash because the client retrieves that value from the server
			(node) => node?.server?.load || node?.server?.trailingSlash !== undefined
```